bucket/drivers/s3/s3.go

```diff
@@ -0,0 +1,113 @@
+package s3
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/sirupsen/logrus"
+)
+
+type S3Driver struct {
+	//svc is IAM S3 service client
+	svc    *s3.S3
+	config *aws.Config
+}
+
+func New(config *aws.Config) *S3Driver {
+	sess, _ := session.NewSession(config)
+	// Create S3 service client
+	svc := s3.New(sess)
+	return &S3Driver{
+		svc:    svc,
+		config: config,
+	}
+}
+
+func (d *S3Driver) CreateBucket(name string) (string, error) {
+	_, err := d.svc.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(name),
+		CreateBucketConfiguration: &s3.CreateBucketConfiguration{
+			LocationConstraint: aws.String(*d.config.Region),
```

Comment:
Region will actually need to a parameter for CreateBucket, as a customer could want a BucketClass for one region and another BucketClass for a different region.

Reply:
As a result, we'll want to create the session/svc for every Create/Delete call that comes in for now. We can do this in a helper function.

Reply:
This will mean adding 3 more parameters to “BucketCreateRequest” param in the OpenStorageBucket service.
Region,
AccessKeyId,
SecretAccessKey
The same 3 parameters will need to be passed to the Bucket and hence BcketDriver interface would need to be updated with the same 3 params.

Two of the above parameters excluding region will need to be passed to the BucketDeleteRequest too.

Can you confirm it is ok to update the Bucket Interface and Request params for Create and Delete  APIs like this?

Reply:
Let's only add the Region parameter to Create/Delete. We can store the AccessKeyID/SecretAccessKey in the s3 driver struct itself.

On bucket create/delete, we can reference the AccessKeyID/SecretAccessKey from the s3 struct.